diff --git a/lib/attach-middleware.js b/lib/attach-middleware.js
index 421541452f00402f08bbdbae2f235ed448abebc2..97fa0f671ad7e74e74ab7f89dc7cac40b6995118 100644
--- a/lib/attach-middleware.js
+++ b/lib/attach-middleware.js
@@ -219,7 +219,9 @@ function reportCoverage(map, root, configPath) {
   });
 
   reporters.forEach((reporter) => {
-    let report = reports.create(reporter, {});
+    let report = Array.isArray(reporter) 
+      ? reports.create(reporter[0], reporter[1])
+      : reports.create(reporter, {});
 
     // call execute to synchronously create and write the report to disk
     report.execute(context);
diff --git a/lib/coverage-merge.js b/lib/coverage-merge.js
index f9328adcfd51fa3f80d524dc6b07c8de21a495cd..1aa8fac66d3327c6ccfb1ea706f565060cbda065 100644
--- a/lib/coverage-merge.js
+++ b/lib/coverage-merge.js
@@ -56,7 +56,9 @@ module.exports = {
           });
 
           reporters.forEach((reporter) => {
-            let report = reports.create(reporter, {});
+            let report = Array.isArray(reporter) 
+              ? reports.create(reporter[0], reporter[1])
+              : reports.create(reporter, {});
 
             // call execute to synchronously create and write the report to disk
             report.execute(context);
